application/models/Survey.php

```diff
@@ -665,7 +665,9 @@ public function getAllLanguages()
     public function getTokenAttributes()
     {
         $attdescriptiondata = decodeTokenAttributes($this->attributedescriptions ?? '');
-
+        if (!is_array(reset($attdescriptiondata))) {
```

Comment:
Why do you call reset() here?

Reply:
Because the reset() function is used by the native code on line 670:
if ($attdescriptiondata && strpos((string) key(reset($attdescriptiondata)), 'attribute_') === false)

And if reset() is not returning an array (in this case a NULL).. key() gives a 500
In trying to debug it is/was very hard to get to this state... strpos(key(reset($attdescriptiondata))... has 3 functions in one line.

The reset(array) is returning null because there are no tokens!

From PHP help:
reset() rewinds array's internal pointer to the first element and returns the value of the first array element.
The value returned by reset(), when not having tokens, is null!

And in PHP<8 it was all good.. but now it is strict! Key(NULL) is not allowed anymore and results in 500

Reply:
key is not allowed to have a string or null as paramater.
In this case the paramater is a empty string!